northbound/query-handler/src/main/java/org/eclipse/sensinact/northbound/query/impl/QueryHandler.java

```diff
@@ -792,15 +789,15 @@ protected Promise<ResponseDescribeResourceDTO> call(SensinactResource resource,
                         dto.type = resource.getResourceType();
                         dto.accessMethods = generateAccessMethodsDescriptions(resource);
                         return resource.getMetadataValues().then(metadata -> {
-                            dto.attributes = generateMetadataDescriptions(resource, metadata.getValue());
+                            dto.attributes = generateMetadataDescriptions(metadata.getValue());
                             return pf.resolved(dto);
                         });
                     }
                 }).then((d) -> {
```

Comment:
Probably not for this PR, but we should be using `map` here as there's no async needed. We should be using `recover` at the end as well.

Reply:
At some point we might need an overall review on Promises handling